src/lib/Libcmds/prepare_path.c

```diff
@@ -190,40 +203,46 @@ prepare_path(char *path_in, char *path_out)
 		 */
 		host_given = NULL;
 		host_name[0] = '\0';
-		path_out[0] = '\0';
-	} else {
+	} else
 #endif
+	{
 		/* get full host name */
 		if (host_name[0] == '\0') {
 			if (pbs_conf.pbs_output_host_name) {
 				/* use the specified host for returning the file */
-				strncpy(host_name, pbs_conf.pbs_output_host_name, PBS_MAXSERVERNAME);
+				strncpy(host_name, pbs_conf.pbs_output_host_name, sizeof(host_name) - 1);
 				have_fqdn = 1;
 			} else {
-				if (gethostname(host_name, PBS_MAXSERVERNAME) != 0)
+				if (gethostname(host_name, sizeof(host_name)) != 0)
 					return 2;
 			}
+			host_name[sizeof(host_name) - 1] = '\0';
```

Comment:
Might want to save the value of 'sizeof(host_name) - 1' since it is repeated reference.

Reply:
Changed the if block to use snprintf() and moved the termination of host_name to the else block.